src/parse/setup_env.c

```diff
@@ -0,0 +1,126 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   setup_env.c                                        :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: walnaimi <walnaimi@student.hive.fi>        +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/08/23 03:25:26 by walnaimi          #+#    #+#             */
+/*   Updated: 2024/08/23 13:25:09 by walnaimi         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "../../includes/minishell.h"
+
+/**
+ * Compares two strings character by character until a mismatch is found.
+ * 
+ * @param s1 The first string to compare.
+ * @param s2 The second string to compare.
+ * 
+ * @return A negative integer if s1 is less than s2, 
+ *         zero if s1 is equal to s2, and a positive integer if s1 is 
+ *         greater than s2.
+ */
+int	ft_strccmp(char *s1, const char *s2)
+{
+	int	i;
+
+	i = 0;
+	while (s1[i] && s1[i] == s2[i])
+		i++;
+	return (s1[i] - s2[i]);
+}
+
+/**
+ * Checks if a given key exists in a linked list of environment variables.
+ * 
+ * @param head A pointer to the head of the linked list of environment variables.
+ * @param word The key to be searched for in the linked list.
+ * 
+ * @return 1 if the key is found, 0 otherwise.
+ */
+int	is_key_in_list(t_env *head, const char *word)
+{
+	while (head != NULL)
+	{
+		if (head->key != NULL && ft_strccmp(head->key, word) == 0)
+		{
+			return (1);
+		}
+		head = head->next;
+	}
+	return (0);
+}
+
+/**
+ * Extracts a word from a given string pointer.
+ *
+ * @param ptr Pointer to the string from which to extract the word.
+ * @param max_word_size Maximum size of the word to be extracted.
+ *
+ * @return A dynamically allocated string containing the extracted word.
+ *
+ * @throws None.
+ */
+char	*extract_word(const char **ptr, int max_word_size)
+{
+	int		index;
+	char	*word;
+
+	word = (char *)malloc((max_word_size + 1) * sizeof(char));
```

Comment:
Malloc check here

Reply:
I do malloc check it just index = 0; in between